contrib/nydusify/pkg/parser/parser.go

```diff
@@ -136,12 +136,22 @@ func (parser *Parser) parseImage(
 	}
 
 	if config.OS == "" || config.Architecture == "" {
-		return nil, errors.New("Source image configuration does not have os or architecture")
+		err = errors.New("Source image configuration does not have os or architecture")
+		if ignoreArch {
+			logrus.WithError(err).Warn("Ignore image arch")
+		} else {
+			return nil, err
+		}
 	}
 
 	// Just give user a simple hint telling option was ignored.
 	if config.Architecture != parser.interestedArch {
-		return nil, errors.Errorf("Specified %s architecture was not found", parser.interestedArch)
+		err = errors.Errorf("Found arch %s, but the specified target arch (--platform) is %s", config.Architecture, parser.interestedArch)
```

Comment:
Should this be move into `else{}` branch?

Reply:
The same error message should be returned if no ignoreArch.